### ISSUE TITLE
Is EHRbase vulnerable to the spring4shell vulnerability

### ISSUE THREAD

**Issue by robert-hunter-nes** [ehrbase/ehrbase#798]
## Question ##

Apologies if this is not the correct place to ask, but I wanted to ask if EHRbase was vulnerable to the active [spring4shell](https://www.lunasec.io/docs/blog/spring-rce-vulnerabilities/) vulnerability?

I'm planning to have a look myself but thought it might be best to put question here in case you have already investigated?

**Comment by maximni-vg** [comment]
Hello @robert-hunter-nes,

thanks for the question. Given the still intransparent state of the issue, I can not give you a full answer yet. We (and the netsec community) are still in analysis of the issue.

We especially monitor the official spring page for the issue: https://spring.io/blog/2022/03/31/spring-framework-rce-early-announcement

As of the time of this comment:
- EHRbase is neither packaged as a WAR, nor ran in an Tomcat, therefore we are not vulnerable to the 0day PoCs that you see in the wild
- Given that this issue is actually very broad with a combined attack based on core concepts of the JVM / Spring container with classloader manipulation and deserializations, it is still unknown what other impact Spring4Shell may have.

What we will do:
- Monitor Spring4Shell closely for further information
- Monitor the Spring page for advice and further information
- If an emergency update will be required, we will provide this ASAP (Spring Boot is not even available yet)

What we will not do:
- We will not include the mitigation controller advice in an emergency release

TL;DR: We're not affected by the main attack vector of this issue, and monitor the issue closely, and will provide an release if required.

**Comment by robert-hunter-nes** [comment]
Thanks @maximni-vg for such a clear and prompt response

**Comment by maximni-vg** [comment]
Hello again,

It seems that there has been lots of misconceptions and FUD going on, given the name similarity to Log4Shell, the similar issue in Spring Cloud Function, and the general information around the issue.

As of now, EHRbase is not affected by Spring4Shell. We will not provide an emergency security release, but update to the latest Spring Boot version with the next release.

As mentioned - if the situation changes, and risks will be identified for non-Tomcat / non-WAR deployments we will provide a release ASAP.

I'll leave this issue open as a tracker for Spring4Shell and EHRbase, until this issue is fully analyzed and remediated.

**Comment by maximni-vg** [comment]
Hello @robert-hunter-nes,

As of now, EHRbase is not affected by Spring4Shell, and the PoC exploit that is available.

We will not provide an emergency security release for this issue, but will still monitor the issue. If no new risks for non-Tomcat / non-WAR scenarios will be identified, we will upgrade the Spring Boot version to the next version in the next regular release, else an emergency release will be provided as soon as possible and required.

I'll leave the issue open for visibility and until the issue has been fully analyzed and remediated.

**Comment by vidi42** [comment]
As latest EHRbase uses Spring 5.3.29 which is not affected by [CVE-2022-22965](https://tanzu.vmware.com/security/cve-2022-22965), we will close the issue
